json.go

```diff
@@ -316,3 +307,53 @@ func appendFloat(buf []byte, v float64, bitSize int) []byte {
 		return strconv.AppendFloat(buf, v, 'f', -1, bitSize)
 	}
 }
+
+// Ref: encodeState#string in encoding/json.
```

Comment:
This implementation might look kind of a horrible idea causing code duplicates, but I don't think there's a better way to solve the original issue with no allocation overhead. I hope there is `json.AppendMarshal([]byte, v any) ([]byte, error)` in `encoding/json` package.

Reply:
Another solution is to create a writer wrapping a `bytes.Buffer` to use `json.Encoder.Encode`, just like I'm proposing in different package (https://github.com/hokaccha/go-prettyjson/pull/21/files#diff-2fc4cc94d53b67bb93f26e56ef7ee4b2344b0b20970b9b971cbf418f4fc31818R233-R244), but this is actually slower than directly appending to `[]byte` for maximum performance like this log package.